File: java/src/main/java/com/experian/dvs/client/address/layout/GetLayoutListItem.java
package com.experian.dvs.client.address.layout;

import com.experian.dvs.client.server.address.layout.RestApiGetLayoutsListItem;

import java.util.List;
import java.util.Objects;

public class GetLayoutListItem {

    private final String id;
    private final String name;
    private final List<AppliesTo> appliesTo;
    private final LayoutStatus status;

    public GetLayoutListItem(RestApiGetLayoutsListItem item) {
        this.id = Objects.requireNonNull(item.getId());
        this.name = Objects.requireNonNull(item.getName());
        this.appliesTo = item.getAppliesTo() != null ? item.getAppliesTo().stream().map(AppliesTo::new).toList() : List.of();
        this.status = item.getStatus() != null ? LayoutStatus.fromName(item.getStatus()) : null;
    }

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public List<AppliesTo> getAppliesTo() {
        return appliesTo;
    }

    public LayoutStatus getStatus() {
        return status;
    }
}

File: csharp/DVSClient/DVSClient/address/layout/GetLayoutListItem.cs
using DVSClient.Common;
using DVSClient.Server.Address.Layout;

namespace DVSClient.Address.Layout
{
    public class GetLayoutListItem
    {
        internal string Id { get; }
        internal string Name { get; }
        internal IEnumerable<AppliesTo> AppliesTo { get; }
        internal LayoutStatus? Status { get; }

        public GetLayoutListItem(RestApiGetLayoutsListItem item)
        {
            Id = item.Id ?? throw new ArgumentNullException(nameof(item.Id));
            Name = item.Name ?? throw new ArgumentNullException(nameof(item.Name));
            AppliesTo = item.AppliesTo != null ? item.AppliesTo.Select(a => new AppliesTo(a)).ToList() : new List<AppliesTo>();
            Status = item.Status?.GetEnumValueFromJsonName<LayoutStatus>();
        }
    }
}